widget/src/providers/ChatProvider.tsx

```diff
@@ -415,13 +419,20 @@ const ChatProvider: React.FC<{
     const endConnection = () => {
       setConnectionState(0);
     };
+    const resetProfile = ({ type, data }: Packet) => {
+      if (type === PacketType.EVENT && data[0] === "settings") {
+        setHasSession(!!data[1]?.hasSession);
+      }
+    };
 
+    socketCtx.socket.io.on("packet", resetProfile);
```

Comment:
Let's re-use the hook that already listens on the settings event

Reply:
The code look's much better after applying the feedback :pray: